pkg/build/builder/source.go

```diff
@@ -77,7 +77,9 @@ func fetchSource(dir string, build *api.Build, urlTimeout time.Duration, in io.R
 func checkRemoteGit(url string, timeout time.Duration) error {
 	glog.V(4).Infof("git ls-remote --heads %q", url)
 	cmd := exec.Command("git", "ls-remote", "--heads", url)
-	cmd.Env = []string{"GIT_ASKPASS=/bin/true", "GIT_SSH=" + os.Getenv("GIT_SSH")}
+	env := os.Environ()
+	env = append(env, "GIT_ASKPASS=/bin/true")
```

Comment:
> I'd like to set up all the envvars once if possible, so they apply equally to ls-remote, clone, and whatever else. I think we're already doing that with GIT_CONFIG, proxy envvars and ssh envvars.

Change coming... actually I need a refactor anyway to leverage the scmauth code outside of the builder (for the gitserver). I'll split my changes from that and use this PR for it.